### PR TITLE
Remove language fallback logic in Environment

### PR DESCRIPTION
**Pull request by njam**
We have the same logic in Request::getLanguage()

@zazabe please review

### CODE REVIEW

File: tests/library/CM/View/DocumentTest.php [hunk 1]
```diff
@@ -100,7 +100,7 @@ public function testLanguageAlternatives() {
         $renderAdapter = new CM_RenderAdapter_Document($render, $page);
         $html = $renderAdapter->fetch();
 
-        $this->assertContains('<link rel="alternate" href="http://www.example.com/en/example" hreflang="x-default">', $html);
```
Comment (njam) [inline tests/library/CM/View/DocumentTest.php]: @zazabe this is actually a bug. The "x-default" variant should link to the version that does auto-detection.

**Comment by zazabe** [comment]
wondering why there was some logic to detect the language in `CM_Frontend_Render` in the past, here:
https://github.com/cargomedia/cm/blob/483b702d8a500b54df8713812080c32bdd3fb68b/library/CM/Frontend/Render.php#L349

File: tests/library/CM/Frontend/RenderTest.php [hunk 2]
```diff
@@ -156,36 +155,6 @@ public function testGetUrlStaticDifferentSite() {
         $this->assertSame('http://cdn.other.com/static/foo.jpg?' . $deployVersion, $render->getUrlStatic('/foo.jpg', $site));
     }
 
-    public function testGetLanguage() {
```
Comment (zazabe) [inline tests/library/CM/Frontend/RenderTest.php]: `$this->_languageDefault` could be removed from `setUp`

**Comment by njam** [comment]
As discussed `Render::getTranslation()` was accessing the default language like that. Instead we'll make sure the Environment always has the default language if needed.

@zazabe please review

File: tests/library/CM/Mail/MailableTest.php [hunk 3]
```diff
@@ -46,9 +46,10 @@ public function testRenderTranslated() {
         $site = $this->getMockSite(null, null, [
             'url' => 'http://www.foo.com',
         ]);
+        $language = CM_Model_Language::create('Test language', 'foo', true);
```
Comment (zazabe) [inline tests/library/CM/Mail/MailableTest.php]: should we add a test with no language for the user?

File: tests/library/CM/Model/UserTest.php [hunk 4]
```diff
@@ -144,4 +144,14 @@ public function testOfflineOld() {
         $this->assertFalse($user1->getOnline());
         $this->assertTrue($user2->getOnline());
     }
+
+    public function testGetEnvironment() {
+        $user = CMTest_TH::createUser();
+        $this->assertNull($user->getEnvironment()->getLanguage());
+
+        $defaultLanguage = CMTest_TH::createLanguage('en');
+        $user = CMTest_TH::createUser();
```
Comment (zazabe) [inline tests/library/CM/Model/UserTest.php]: maybe add a test when a user has no language set, but he get the default one on his env?

Comment (njam) [inline tests/library/CM/Model/UserTest.php]: It's covered 2 lines below I think

**Comment by njam** [comment]
@zazabe please review

Could you take over here and release it? We should do it soon to make sure we don't run into problems with Google indexing.

File: tests/library/CM/Model/UserTest.php (hunk 4 above)
Comment (zazabe) [inline tests/library/CM/Model/UserTest.php]: not really, it should be ` $this->assertEquals($defaultLanguage, $user->getEnvironment()->getLanguage());`, but without setting `$user->setLanguage($defaultLanguage)`, as below...

**Comment by zazabe** [comment]
yep!

File: tests/library/CM/Model/UserTest.php (hunk 4 above)
Comment (zazabe) [inline tests/library/CM/Model/UserTest.php]: like 74a211a

Comment (njam) [inline tests/library/CM/Model/UserTest.php]: Makes sense

**Pull request merged**